src/Log/LogServiceTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * This file is part of OPUS. The software OPUS has been originally developed
+ * at the University of Stuttgart with funding from the German Research Net,
+ * the Federal Department of Higher Education and Research and the Ministry
+ * of Science, Research and the Arts of the State of Baden-Wuerttemberg.
+ *
+ * OPUS 4 is a complete rewrite of the original OPUS software and was developed
+ * by the Stuttgart University Library, the Library Service Center
+ * Baden-Wuerttemberg, the Cooperative Library Network Berlin-Brandenburg,
+ * the Saarland University and State Library, the Saxon State Library -
+ * Dresden State and University Library, the Bielefeld University Library and
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Test
+ * @package     Opus
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace Opus\Log;
+
+use Opus\Log\LogService;
+
+class LogServiceTest extends \PHPUnit_Framework_TestCase
+{
+    private $logInstance;
+    private $logger;
+
+    public function setUp()
+    {
+        $this->logInstance = LogService::getInstance();
+        $this->logInstance->setPath('opus.log');        //may also mention new path as a second parametre
+        $this->logInstance->openLog('Zend_Config');
+
+        $this->logger = $thisLogInstance->setLog();    //how to get comparable log
+    }
+
+    public function testGetInstance()
+    {
+        $logInstance = new LogService();
```

Comment:
LogService should never be used this way. It is meant to be a singleton, so it should not be possible to use __new__. You can suppress that by having a private/protected __construct function. We haven't used that in PHP very much yet, so I do not know if there are any surprises waiting, but generally in OOP it should work that way.

Reply:
And you would expect a `testGetInstance` function to test `getInstance`.

Reply:
Yes, I know it shouldn't be this way. Again, this commit wasn't meant to be reviewed, I left it like this just for me as a question, like, what if I do this, will correct it before a final commit for review. Sorry again for the confusion.